lib/rspec/core/formatters/failure_list_formatter.rb

```diff
@@ -17,6 +27,55 @@ def example_failed(failure)
         # list formatter.
         def dump_profile(_profile); end
         def message(_message); end
+
+        private
+
+        # @return [Array<Exception>]
+        def exceptions(failure)
+          if failure.exception.respond_to?(:all_exceptions)
+            failure.exception.all_exceptions
+          else
+            [failure.exception]
+          end
+        end
+
+        # @return [String]
+        def one_line(exception)
+          exception.message.lines.map(&:strip).reject(&:empty?).join ' '
+        end
+
+        # @return [Array<String>, nil] relevant location with relative path, if any
+        def find_locations(exception)
+          require 'pathname'
```

Comment:
Sorry but we don't want to include the standard library like this, plus the formatter is showing the example that failed, it doesn't need to be the exception location.

Reply:
Honestly, it wasn't my first choice. I wanted to include it the regular way, but another test failed, complaining about the set of set of `require`s that RSpec brings with itself. I moved it back up. Expect to see failed checks.

> plus the formatter is showing the example that failed, it doesn't need to be the exception location.

Could you rephrase that? I don't understand.

Reply:
This is the failure that occurs with a regular `require` at the top of the file:

```
Failures:
99

100
  1) RSpec behaves like library wide checks only loads a known set of stdlibs so gem authors are forced to load libs they use to have passing specs
101
     Failure/Error: @example_group_instance.instance_exec(self, &@example_block)
102

103
       expected: []
104
            got: ["/opt/hostedtoolcache/Ruby/2.5.8/x64/lib/ruby/2.5.0/x86_64-linux/pathname.so", "/opt/hostedtoolcache/Ruby/2.5.8/x64/lib/ruby/2.5.0/pathname.rb"]
105

106
       (compared using ==)
107
     Shared Example Group: "library wide checks" called from ./spec/rspec/core_spec.rb:31
108
     # /home/runner/work/rspec-core/rspec-support/lib/rspec/support.rb:97:in `block in <module:Support>'
109
     # /home/runner/work/rspec-core/rspec-support/lib/rspec/support.rb:106:in `notify_failure'
110
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/fail_with.rb:35:in `fail_with'
111
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/handler.rb:38:in `handle_failure'
112
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/handler.rb:56:in `block in handle_matcher'
113
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/handler.rb:27:in `with_matcher'
114
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/handler.rb:48:in `handle_matcher'
115
     # /home/runner/work/rspec-core/rspec-expectations/lib/rspec/expectations/expectation_target.rb:65:in `to'
116
     # /home/runner/work/rspec-core/rspec-support/lib/rspec/support/spec/library_wide_checks.rb:129:in `block (2 levels) in <top (required)>'
117
     # ./lib/rspec/core/example.rb:262:in `instance_exec'
118
     # ./lib/rspec/core/example.rb:262:in `block in run'
119
     # ./lib/rspec/core/example.rb:508:in `block in with_around_and_singleton_context_hooks'
120
     # ./lib/rspec/core/example.rb:465:in `block in with_around_example_hooks'
121
     # ./lib/rspec/core/hooks.rb:486:in `block in run'
122
     # ./lib/rspec/core/hooks.rb:626:in `block in run_around_example_hooks_for'
123
     # ./lib/rspec/core/example.rb:350:in `call'
124
     # ./spec/support/sandboxing.rb:16:in `block (3 levels) in <top (required)>'
125
     # ./lib/rspec/core/sandbox.rb:29:in `sandboxed'
126
     # ./spec/support/sandboxing.rb:7:in `block (2 levels) in <top (required)>'
127
     # ./lib/rspec/core/example.rb:455:in `instance_exec'
128
     # ./lib/rspec/core/example.rb:455:in `instance_exec'
129
     # ./lib/rspec/core/hooks.rb:390:in `execute_with'
130
     # ./lib/rspec/core/hooks.rb:628:in `block (2 levels) in run_around_example_hooks_for'
131
     # ./lib/rspec/core/example.rb:350:in `call'
132
     # ./lib/rspec/core/hooks.rb:629:in `run_around_example_hooks_for'
133
     # ./lib/rspec/core/hooks.rb:486:in `run'
134
     # ./lib/rspec/core/example.rb:465:in `with_around_example_hooks'
135
     # ./lib/rspec/core/example.rb:508:in `with_around_and_singleton_context_hooks'
136
     # ./lib/rspec/core/example.rb:259:in `run'
137
     # ./lib/rspec/core/example_group.rb:644:in `block in run_examples'
138
     # ./lib/rspec/core/example_group.rb:640:in `map'
139
     # ./lib/rspec/core/example_group.rb:640:in `run_examples'
140
     # ./lib/rspec/core/example_group.rb:606:in `run'
141
     # ./lib/rspec/core/example_group.rb:607:in `block in run'
142
     # ./lib/rspec/core/example_group.rb:607:in `map'
143
     # ./lib/rspec/core/example_group.rb:607:in `run'
144
     # ./lib/rspec/core/runner.rb:121:in `block (3 levels) in run_specs'
145
     # ./lib/rspec/core/runner.rb:121:in `map'
146
     # ./lib/rspec/core/runner.rb:121:in `block (2 levels) in run_specs'
147
     # ./lib/rspec/core/configuration.rb:2067:in `with_suite_hooks'
148
     # ./lib/rspec/core/runner.rb:116:in `block in run_specs'
149
     # ./lib/rspec/core/reporter.rb:74:in `report'
150
     # ./lib/rspec/core/runner.rb:115:in `run_specs'
151
     # ./lib/rspec/core/runner.rb:89:in `run'
152
     # ./lib/rspec/core/runner.rb:71:in `run'
153
     # ./lib/rspec/core/runner.rb:45:in `invoke'
154
     # ./exe/rspec:4:in `<top (required)>'
155
     # script/rspec_with_simplecov:42:in `load'
156
     # script/rspec_with_simplecov:42:in `<main>'
```